### PR TITLE
DOC: fail the CI build and do not deploy if docs are not created

### LINKED ISSUE #23063: DOCS: deployment of devdocs failed, deleting the docs
https://github.com/numpy/devdocs is showing a 404. This is because the deployment step (after a merge-to-main) is removing the docs from https://github.com/numpy/devdocs instead of uploading them. [The deploy step](https://github.com/numpy/numpy/blob/747823cc7e77d1e13705856d10a6ca2a46f4c47d/.circleci/config.yml#L128) should not be run if the build is empty.

This is due to the [documentation build](https://app.circleci.com/pipelines/github/numpy/numpy/17729/workflows/c43d2c87-e712-44ae-9363-e911378bdf01/jobs/29775?invite=true#step-109-172) failing but not reporting the error. Probably fallout from #22948 unifying the two doc build jobs. @tupui could you take a look?

### PR DESCRIPTION
**Pull request by mattip**
Fixes #23063 by
- checking that the sphinx build actually output some files
- refuse to deploy if a minimum count of files is not found

### CODE REVIEW

File: tools/ci/push_docs_to_repo.py [hunk 1]
```diff
@@ -31,6 +33,11 @@
     print('Content directory does not exist')
     sys.exit(1)
 
+count = len([name for name in os.listdir(args.dir) if os.path.isfile(os.path.join(args.dir, name))])
```
Comment (mattip) [inline tools/ci/push_docs_to_repo.py:36]: Hmm, I think I like this better than splitting it over two lines, even though linting objects

**Comment by mattip** [comment]
Docs build failed as expected since #23064 was not merged in yet. Now I will rebase and it should pass.

**Comment by mattip** [comment]
Doc building passes.

We are on the edge: 200 open PRs. Merging this will get us down to 199!

**Review by rossbar — APPROVED** [review]
Seems fine to me as a temporary fix, though it's a little annoying that we have to hand-engineer another check to prevent this rather than the deploy job just failing when the doc-build fails (shouldn't the `requires: build` be enough? Why is this passing?)

**Comment by mattip** [comment]
This is double-protection. The changes to the circleci.yaml will fail the build if sphinx fails, even in nitpick mode. The changes to the deploy script will fail to deploy even if a mis-configured CI improperly builds the docs and does not fail.

**Pull request merged**